metabase/metabase
# Preserve saved parameter operator type

Closes https://github.com/metabase/metabase/issues/41658

### Description

If use changes filter type for the saved dashboard and saved parameter mapping and then switches back to the saved parameter type, we should restore the saved parameter value.

UX might be not the best (I presume showing a link or a button to reset state would be better), but lets see how it goes

### How to verify

- create a dashboard with a filter and question (orders table and text filter work)
- create a mapping between card and filter, set operator to "does not contain" and label to "text 123", save dashboard
- edit dashboard, set another parameter type (e.g. number), then change again to Text or Category -> label and operator should be restored to the saved values

### Demo



https://github.com/metabase/metabase/assets/125459446/d51aa776-a0b2-4232-80ae-f6997fb9670b





### Checklist

- [x] [stress test](https://github.com/metabase/metabase/actions/runs/8909448663/job/24466830272)
- [x] [stress test for second test](https://github.com/metabase/metabase/actions/runs/8925151406)
